backend/src/LibExecutionStdLib/LibList.fs

```diff
@@ -1388,6 +1388,138 @@ let fns : List<BuiltInFn> =
       previewable = Impure
       deprecated = NotDeprecated }
 
+
+    { name = fn "List" "groupAdjacentBy" 0
+      typeParams = []
+      parameters =
+        [ Param.make "list" (TList varA) ""
+          Param.makeWithArgs "fn" (TFn([ varA; varA ], TBool)) "" [ "a"; "b" ] ]
+      returnType = TList(TList varA)
+      description =
+        "Groups <param list> into sublists, where each sublist contains values that
+         are equal according to <param fn>.
+
+         For example, if <param list> is {{[1, 2, 2, 3, 2, 2, 2, 1, 1, 1]}} and <param fn>
```

Comment:
This is a sort of odd function and I agree that it's unclear what someone would use it for.

I think perhaps you included this because of my comments about wanting a function wiht the signature `List<'a> -> ('a -> 'a -> bool) -> List<List<'a>>`. I think I may have been mistaken here. What I meant to say was a function whose return type was `List<List<'a>>` - I wasn't actually that concerned with the predicate.

What I had intended to suggest was

`List.groupBy: List<'a> -> ('a -> 'key) -> List<List<'a>>`
and
`List.groupByWithKey: List<'a> -> ('a -> 'key) -> List<'key * List<'a>>`

(I think a `Dict<'key, List<'a>>` would be better but we don't support that yet.)

Reply:
> perhaps you included this because of my comments about wanting a function wiht the signature List<'a> -> ('a -> 'a -> bool) -> List<List<'a>>.

yes, and I figured it's based on Haskell, since that's what I consider the de facto standard for anything functional programming

I assume the `groupAdjacentBy` could be conformable for someone accustomed to Haskell

I don't mind renaming the `groupBy` to `groupWithKeyBy`. 

Regarding `List.groupBy: List<'a> -> ('a -> 'key) -> List<List<'a>>`, wouldn't it be better to implement it in Dark since it would require the `groupByWithKey` and a map

Reply:
Yeah, good point.